crates/extensions/tedge_mqtt_ext/src/lib.rs

```diff
@@ -0,0 +1,207 @@
+#[cfg(test)]
+mod tests;
+
+use async_trait::async_trait;
+use mqtt_channel::MqttError;
+use mqtt_channel::SinkExt;
+use mqtt_channel::StreamExt;
+use mqtt_channel::TopicFilter;
+use tedge_actors::mpsc;
+use tedge_actors::mpsc::channel;
+use tedge_actors::mpsc::Receiver;
+use tedge_actors::mpsc::Sender;
+use tedge_actors::Actor;
+use tedge_actors::ActorBuilder;
+use tedge_actors::ChannelError;
+use tedge_actors::DynSender;
+use tedge_actors::LinkError;
+use tedge_actors::MessageBox;
+use tedge_actors::RuntimeError;
+use tedge_actors::RuntimeHandle;
+use tedge_actors::SimpleMessageBox;
+
+pub type MqttConfig = mqtt_channel::Config;
+pub type MqttMessage = mqtt_channel::Message;
+
+#[derive(Debug)]
+pub enum MqttActorMessage {
+    DirectMessage(mqtt_channel::Message),
+    PeerMessage(mqtt_channel::Message),
+}
```

Comment:
I would definitely get rid of this envelop type and deal only with the peer messages.

There are 2 reasons for an actor to expose an enum type with all the message types (requests, responses, events, reactions, ...): 
1. To interact with other actors sending or receiving some of these messages.
2. To test the actor, feeding it with arbitrary sequences of input, and observing all its output.

In that case:
1. Not a single actor peer will emit or listen to `DirectMessage`
2. Injecting `DirectMessage` from a test into the actor makes no sense.

So I would only expose the `PeerMessage`:

```
impl MessageBox for MqttMessageBox {
    type Input = MqttMessage;
    type Output = MqttMessage;

    ...
```